### PR TITLE
allow adding docker Private registry details while deploying cluster

### PR DESCRIPTION
**Pull request by Pearl1594**
Addresses the following:
1. Allows users to add private registry - if experimental features flag is enabled
2. Sets min number of master nodes to 2 if HA is enabled

Added the Private registry flag:
image

For HA, made the default value of number of master nodes to 2:
Default initial value:
image

Setting it to a value lesser than 2 (which ideally isn't HA), will prompt the user:
image

### CODE REVIEW

**Comment by davidjumani** [comment]
Fixes https://github.com/apache/cloudstack-primate/issues/801

**Comment by Pearl1594** [comment]
@blueorangutan package

File: src/views/compute/CreateKubernetesCluster.vue [hunk 1]
```diff
@@ -307,6 +366,7 @@ export default {
   },
   methods: {
     fetchData () {
+      console.log(this.$store.getters.features)
```
Comment (shwstppr) [inline src/views/compute/CreateKubernetesCluster.vue:369]: not really needed

**Review by shwstppr — APPROVED** [review]
LGTM except for one minor nit

**Comment by Pearl1594** [comment]
@blueorangutan package

File: src/components/view/ListView.vue [hunk 2]
```diff
@@ -439,7 +439,7 @@ export default {
         value: this.editableValue
       }).then(json => {
         this.editableValueKey = null
-
+        this.$store.dispatch('RefreshFeatures')
```
Comment (rhtyd) [inline src/components/view/ListView.vue:442]: @Pearl1594 why are we doing this here? Assuming we do list capabilities fetching already on log in?

Comment (Pearl1594) [reply src/components/view/ListView.vue:442]: @rhtyd If this isn't done, then unless one logs out and logs back in or does a hard refresh, the store isn't going to be updated with the new value of 'kubernetesclusterexperimentalfeaturesenabled' in the capabilities list

File: src/store/modules/user.js [hunk 3]
```diff
@@ -274,6 +274,16 @@ const user = {
           reject(error)
         })
       })
+    },
+    RefreshFeatures ({ commit }) {
+      return new Promise((resolve, reject) => {
+        api('listCapabilities').then(response => {
+          const result = response.listcapabilitiesresponse.capability
+          commit('SET_FEATURES', result)
+        }).catch(error => {
```
Comment (davidjumani) [inline src/store/modules/user.js:283]: @Pearl1594 Would not resolving the promise cause any issues ? (Promise state, GC, etc)

**Comment by Pearl1594** [comment]
@blueorangutan package

**Review by shwstppr — APPROVED** [review]
lgtm

**Pull request merged**